api/api.go

```diff
@@ -252,13 +253,11 @@ func (api *Server) GetChainMeta(ctx context.Context, in *iotexapi.GetChainMetaRe
 		return nil, status.Error(codes.NotFound, err.Error())
 	}
 
-	timeDuration := blks[len(blks)-1].Timestamp.GetSeconds() - blks[0].Timestamp.GetSeconds()
-	// if time duration is less than 1 second, we set it to be 1 second
-	if timeDuration < 1 {
-		timeDuration = 1
-	}
-
-	tps := numActions / timeDuration
+	t1 := time.Unix(blks[0].Timestamp.GetSeconds(), int64(blks[0].Timestamp.GetNanos()))
+	t2 := time.Unix(blks[len(blks)-1].Timestamp.GetSeconds(), int64(blks[len(blks)-1].Timestamp.GetNanos()))
+	// duration of tps window in milli-seconds
+	timeDuration := t2.Sub(t1) / time.Millisecond
```

Comment:
time duration will be 0 if len(blks) is 1. It is a corner case, which rarely happens.

Reply:
fixed in latest push